presto-main/src/main/java/com/facebook/presto/sql/planner/SymbolsExtractor.java

```diff
@@ -106,6 +106,15 @@ private SymbolsExtractor() {}
         return ImmutableSet.copyOf(extractAll(expression));
     }
 
+    public static Set<VariableReferenceExpression> extractUniqueVariable(Iterable<? extends RowExpression> expressions)
```

Comment:
Doesn't this overlap with `public static Set<VariableReferenceExpression> extractUniqueVariable(Iterable<? extends Expression> expressions, TypeProvider types)`?

Reply:
Yes, but this one assumes `planNode` does not contain `OriginalRowExpression`. As we move translation rule up, we should change to use `extractUniqueVariable(Iterable<? extends Expression> expressions)` gradually. 

`public static Set<VariableReferenceExpression> extractUniqueVariable(Iterable<? extends Expression> expressions, TypeProvider types)` probably should be marked as `deprecated`.